### ISSUE TITLE
Submitting Data via text input crashes universal windows app

### ISSUE THREAD

**Issue by jb-207** [microsoft/react-native-windows#8883]
I have a small 3 field form and when I try and press enter (or any key) to run a handleSubmit function. It crashes the app with the code in Visual Studio: Unhandled exception at 0x00007FF9C0005DC6 (Microsoft.ReactNative.dll) in appname.exe: 0x00000000: The operation completed successfully (parameters:0x0000000000000001, 0x0000000000000000). This occurs in the crash_min.cpp file.

When I press a button that runs the exact same function it runs as expected.

Here is the the code with the form:
```jsx
 handleKeyDown(event) {
    if(event.nativeEvent.key == "Enter"){
     this.submitBtnRef.props.onPress()
  }
}

 return (
      <View style={App_Styles.subContainer}>
         <TouchableWithoutFeedback onPress={() => this.props.linkOpenType(this.props.state.global_styles.site)}>
           <View style={App_Styles.image_link}>
            <Image_Output 
              style={App_Styles.logo_hero}
              source={{uri: this.props.state.global_styles.logo}}
              desiredWidth='525'
              desiredHeight='525'
            />
            </View>
          </TouchableWithoutFeedback>
      {welcome_msg}
      {this.props.state.code != '' ? <AppText style={subtitle_styles}>{this.props.state.status_msg}</AppText> : null}
      <AppText style={[App_Styles.label,{marginTop:10}]}>{this.props.state.global_styles.domain_name_placeholder} (e.g. yoursite.com)</AppText>
      <TextInput
        style={App_Styles.input}
        onChange={this.props.handleChange('domain_name')}
        value={this.props.state.domain_name}
        placeholder={this.props.state.global_styles.domain_name_placeholder}
        autoFocus={true}
        onKeyPress={(event) => {this.handleKeyDown(event)}}

      />
      <AppText style={App_Styles.label}>Email</AppText>
      <TextInput
        style={App_Styles.input}
        onChange={this.props.handleChange('email')}
        value={this.props.state.email}
        placeholder="Email"
      />
      <Text style={App_Styles.label}>Password</Text>
        <View style={{position:"relative"}}>
          <TextInput
              style={[App_Styles.input,{paddingRight:45}]}
              onChange={this.props.handleChange('password')}
              value={this.props.state.password}
              placeholder="Password"
              secureTextEntry={(this.props.state.pw_field_type == 'password') ? true : false}
          />
          <TouchableOpacity style={{position:"absolute",right:11,top:14}} onPress={() => this.props.showHidePW()}>
              <View>
                <Icon name={this.props.state.pw_field_icon} size={27} color={icon_color} />
              </View>
          </TouchableOpacity>
        </View>
      <Virt_Button ref={this.submitBtnRef} state={this.props.state} style={{padding:"14,10,16,10",margin:"10,0,0,0",fontSize:35}} showLoading={this.props.showLoading} onPress={() => {this.props.handleSubmit('credentials_form')}}>Submit</Virt_Button>
      {this.props.state.loading_icon === true ? <Loader_Icon /> : null}
      </View>
    )
```

And the parent Function that is passed down to this component:
```jsx
handleSubmit(formname) {
    console.log('testSubmit')
    let domain_arr = this.state.domain_name.split('//')
    domain_arr = domain_arr.pop()
    domain_arr = domain_arr.split('www.')
    let domain_cleaned = domain_arr.pop()
    this.setState({
      loading_icon:true,
      domain_cleaned:domain_cleaned
    })
     if (formname == 'credentials_form') {
      if (this.state.email != '' && this.state.password != '' &&  this.state.domain_name) {
          fetch(this.state.domain_prefix+domain_cleaned+"/api/module_name/login/"+this.state.email+"/purejson/"+this.state.password, {
            method:'GET',
            headers: {
              'Accept': this.state.header_accept,
              'Content-Type': this.state.header_content_type
            }
          })
          .then(async response => {
            try {
            const data = await response.json()
            return data
          } catch(error) {
            console.log('error happened here after submit.')
            console.log(error)
          }
        }).then(response => {
            if (response[this.state.global_styles.status_keyword] == this.state.global_styles.success_keyword) {
              const value = {
                refresh_token:response.refresh_token,
                domain_cleaned:domain_cleaned,
                email:response.visitor_email,
                loggedOut:false,
                firstTimeLogin:false
              }
              this.setState({
                  refresh_token:response.refresh_token,
                  loading_icon:false,
                  login_error:false,
                  visitor_email:this.state.email,
                  screen:'show_orders',
                  isLoading:false,
                  login_error:false,
                  pw_field_type:"password",
                  status_msg:this.state.global_styles.successful_login_msg,
                  status:response.status,
              })
            } else {
              const screen_choice = (response.code != '' ? 'enter_credentials' : 'error')
              this.setState(() => {
                return {
                  status_msg:response.message,
                  screen:screen_choice,
                  status:response.status,
                  loading_icon:false,
                  login_error:true
                }
              })
              this.revertInitialMessage()
            }
          }).catch(err => {
            this.setState({
                status_msg:this.state.fetch_login_err_msg,
                screen:"enter_credentials",
                status:'',
                isLoading:false,
                loading_icon:false,
                login_error:true
            })
            console.log('Check Your Credentials')
            console.log(err)
        })
      } else {
        this.setState(() => {
          return {
            status_msg:this.state.initial_msg,
            screen:"enter_credentials",
            status:'',
            loading_icon:false,
            login_error:false
          }
        })
      }
    } else {
      this.setState({
        loading_icon:false,
        login_error:false
      })
    } 
  }
```
Any thoughts?

Thanks!

**Comment by acoates-ms** [comment]
I know we still havent gotten a crash dump with real info.  But I also suspect that this crash might be fixed in 0.66.3.  I think the issue could be the same as https://github.com/microsoft/react-native-windows/pull/8973.

**Comment by creambyemute** [comment]
@acoates-ms I would have provided a crash dump but since the above mentioned PR was merged into 0.66/0.65 I haven't experienced any crashes on TextInput onSubmit anymore.

**Comment by acoates-ms** [comment]
Sounds like this issue is resolved then.
